classes/battle/BattleManager.php

```diff
@@ -54,6 +56,7 @@ public function __construct(System $system, User $player, int $battle_id, bool $
         $this->player = $player;
         $this->spectate = $spectate;
         $this->battle = new Battle($system, $player, $battle_id);
+        $this->retreat = $this->battle->is_retreat;
```

Comment:
Huh, why not the same name here?
```suggestion
        $this->is_retreat = $this->battle->is_retreat;
```